### DISCUSSION TITLE
Question: Accessing same database from different client APIs

### DISCUSSION THREAD

**Discussion by amoghnjoshi** [duckdb/duckdb#3448] [Q&A] [answered]
For the same file backed DuckDB instance, is it a valid use-case to insert data with c++ client APIs and query the same from python APIs?  In other words, is the serialization compatible across client APIs?

**Comment by SartorialOffense** [comment, marked as answer]
Not a DuckDB expert, but the client API's are just wrappers and delegate persistence to the underlying framework which is C++. You can insert data using Python, mutate via the CLI interface, retrieve the results using the C API and it should all work.

**Reply by amoghnjoshi** [reply]
Thanks SartorialOffense. 
I was expecting it to be like that. However, there was a 'serialization error - Attempting to read a required field, but field is missing' observed while making a python connection to a JDBC created database.
Upon more investigation, it turned out to be a version mismatch between python and JDBC client versions (0.3.3 vs 0.3.1). The problem got fixed after using same version (0.3.1) for both the clients.
